### PR TITLE
ci: Build and Push this package to CodeArtiact

### PR DESCRIPTION
**Pull request by loydbanks**

### CODE REVIEW

**Review by paddatrapper — CHANGES_REQUESTED** [review]
This publishes every time a push to master happens, but the version is always the same because it isn't changed on push. This needs more configuration. Either versioning is manual and this gets run on tags or versioning needs to follow semantic versioning and this gets run as part of the semantic versioning process

**Comment by paddatrapper** [comment]
Please use the same sort of semantic release process we use elsewhere - https://github.com/ClearlyEnergy/BEAM/blob/develop/.github%2Fworkflows%2Fhelm-chart.yml

There are several steps:

1. Run semantic release (with appropriate release.yml configuration file)
2. If there is a new release update pyproject.toml with the new version number
3. Build the python package
4. Publish to codeartifact

File: .github/worfklows/codeartifact-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Download yq
+        run: |
+          wget -nv -nc -O yq https://github.com/mikefarah/yq/releases/download/v4.20.2/yq_linux_amd64
+          chmod +x yq
+      - name: Update pyproject.toml version
+        env:
+          RELEASE_VERSION: ${{ needs.semantic-release.outputs.release-version }}
+        run: ./yq -i e '.version = "${{ env.RELEASE_VERSION }}"' pyproject.toml
```
Comment (paddatrapper) [inline .github/worfklows/codeartifact-deployment.yml:38]: Yq won't work here - it's not a yaml file

File: .github/worfklows/codeartifact-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+      - name: Push changes to beam_opt repo
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: "Release ${{ needs.semantic-release.outputs.release-version }} of ${{ github.repository }}"
+          branch: gh-pages
+          repository: ./beam_opt
+
+  publish:
```
Comment (paddatrapper) [inline .github/worfklows/codeartifact-deployment.yml:47]: This shouldn't be a separate job - it should only happen when there is a release

File: .github/worfklows/codeartifact-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Download yq
+        run: |
+          wget -nv -nc -O yq https://github.com/mikefarah/yq/releases/download/v4.20.2/yq_linux_amd64
+          chmod +x yq
+      - name: Update pyproject.toml version
+        env:
+          RELEASE_VERSION: ${{ needs.semantic-release.outputs.release-version }}
+        run: ./yq -i e '.version = "${{ env.RELEASE_VERSION }}"' pyproject.toml
+
+      - name: Push changes to beam_opt repo
```
Comment (paddatrapper) [inline .github/worfklows/codeartifact-deployment.yml:40]: Remove this step - we're not pushing to another repo

**Review by paddatrapper — CHANGES_REQUESTED** [review]

File: .github/worfklows/codeartifact-deployment.yml (hunk 1 above)
Comment (loydbanks) [reply .github/worfklows/codeartifact-deployment.yml:38]: https://kislyuk.github.io/yq/#toml-support, they do support toml on Yq

**Review by paddatrapper — APPROVED** [review]

**Pull request merged**